src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -490,7 +490,7 @@ public abstract partial class JsonConverter<T> : System.Text.Json.Serialization.
         protected internal JsonConverter() { }
         public override bool CanConvert(System.Type typeToConvert) { throw null; }
         public abstract T Read(ref System.Text.Json.Utf8JsonReader reader, System.Type typeToConvert, System.Text.Json.JsonSerializerOptions options);
-        public abstract void Write(System.Text.Json.Utf8JsonWriter writer, [System.Diagnostics.CodeAnalysis.DisallowNull] T value, System.Text.Json.JsonSerializerOptions options);
+        public abstract void Write(System.Text.Json.Utf8JsonWriter writer, T value, System.Text.Json.JsonSerializerOptions options);
```

Comment:
Are we sure we want to go ahead with this change without the accompanying https://github.com/dotnet/runtime/issues/34439?

See https://github.com/dotnet/runtime/pull/32186#discussion_r379162610

Would existing converters that don't currently check for null input need to be fixed up to handle that?

> Our internal converts pass null (String converter for one)

The StringConverter is explicitly annotated to allow nullable string. Do we have other examples?

https://github.com/dotnet/runtime/blob/c27376e46ccec555b00eb45cefaa1b3712b44a99/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/StringConverter.cs#L7

Reply:
> Are we sure we want to go ahead with this change without the accompanying #34439?
I think so. Even without that, the String converter gets passed null (any internal converter that uses a reference type). Does it make sense to have `DisallowNull` since we override the behavior internally?

Also, for converter composition where one converter calls another converter there may be a desire to pass null as the value -- especially if the target converter does something special with null. 

> Would existing converters that don't currently check for null input need to be fixed up to handle that?

There is no behavior change with the PR. When #34439 is implemented, null will not be passed to any converters (except internal) unless they opt-in through the new virtual method.